cmd/new.go

```diff
@@ -166,11 +166,16 @@ func createAndEditSnippet(newSnippet snippet.SnippetInfo, snippets snippet.Snipp
 }
 
 func countSnippetLines() int {
-	// Count lines in snippet file
-	f, err := os.Open(config.Conf.General.SnippetFile)
+	filepath, err := config.ExpandPath(config.Conf.General.SnippetFile)
+	if err != nil {
+		return 0
+	}
```

Comment:
As mentioned in #321, I made an assumption that if SnippetFile is omitted, it should simply return 0 for line count instead of panicking. 

Perhaps whoever understands multiple directories better could correct me here.

Reply:
hmmm yes let me look into this further. 

I'm concerned about having a relative snippet file with directories, not sure how that would be handled.
That's even a concern in the current version of pet where you might have an absolute snippet file path as well as directories. Might not be an issue, just have to look into it and possibly document it.

Reply:
I think having both `SnippetDirs` and `SnippetFile` configurable can be rather confusing due to the lack of signs to indicate which one of those two options should take priority.

I'd propose something like this:

```
SnippetFiles = ["~/.config/pet/snippet-1.toml", "/var/tmp/pet/snippet-2.toml" ]
```

This consolidates the option of managing multiple directories under one single configurable item. 
If you want to keep everything under a single file, specify a single snippet file.
If you want to manage multiple files, in various directories, specify multiple paths. 
When save/edit, a prompt to choose which file to save/edit is simple enough. 

Of course, I image it'd be a breaking change.

Reply:
@RamiAwar I could take a jab at this approach later this week if you think it's worth considering

Reply:
@csessh sorry I still didn't look into this more, I was actually typing up a reply at some point but lost the tab 😓 

I think it's not a problem and there's no work necessary on your end, but I want to better document this. Will respond in a few days time!

Reply:
Going to add some more context here - will add it to the docs later as well. You don't have to act on this, just FYI, going through this to review your PR properly.

Having the main snippet file is mandatory.

If snippet directories are also configured, then the snippet Load method will load snippets from the main file + all files inside the snippet directory!

**What does this mean for:**
**1- Edits:** How do we edit a snippet if it can be part of any of several files? -> We make the user select the snippet they want to edit first, which tells us which file we should open for editing.

This is still not amazing UX because we send them to the beginning of the file and not the correct snippet line - this should be fixed. (should add a test for this, making sure we jump to the right line + fix it)

2- Search: We just search across all snippets from all files.

More fundamental operations:

3- Load: Loads snippets from the core file (mandatory) + all the files in all directories specified into one big list. Nice, expected.

4- Save: Given a slice of SnippetInfo, this loops over every snippet and checks its filename. This logic doesn't look right to me - we're just overwriting the snippetFile as we loop through? If it's empty, we're setting it to the FIRST snippet directory + a custom file name (???). 

I think here it was meant to be `snippet.Filename` maybe. This definitely seems like a bug, consistent with what you reported in that issue #321 I think.

<img width="660" alt="image" src="https://github.com/user-attachments/assets/8ec7a1b7-6fba-4bb8-b88d-4d42fe65f244">

Reply:
Do you think it makes sense for the sync to only sync the main snippet file and ignore files in the directories?

I want to try and avoid having to sync multiple files and keep track of each. Directories are nice, but I want to keep it a 'secondary' feature until I iron out the kinks. There are quite some bugs

Reply:
#323 fixes some of these

Reply:
It'll be fine if it's communicated clearly in the tool's documentation that only main snippet file is sync'd. 
I, personally, don't use directories. sync or not.

Reply:
> As mentioned in #321, I made an assumption that if SnippetFile is omitted, it should simply return 0 for line count instead of panicking.
> 
> Perhaps whoever understands multiple directories better could correct me here.

Back to this - I think this needs a rebase now. New code actually panics here, which is what we want. Snippet file should always be present